src/lava/proc/sparse/models.py

```diff
@@ -0,0 +1,359 @@
+            sign_mode: SignMode = self.proc_params.get("sign_mode") \
+                or determine_sign_mode(self.weights)
+
+            self.weights = clip_weights(self.weights, sign_mode, num_bits=8)
+            self.weights = truncate_weights(self.weights,
+                                            sign_mode,
+                                            num_weight_bits)
+            self.weights_set = True
+
+        # The a_out sent at each timestep is a buffered value from dendritic
+        # accumulation at timestep t-1. This prevents deadlocking in
+        # networks with recurrent connectivity structures.
+        self.a_out.send(self.a_buff)
+        if self.num_message_bits.item() > 0:
+            s_in = self.s_in.recv()
+            a_accum = self.weights.dot(s_in)
+        else:
+            s_in = self.s_in.recv().astype(bool)
+            a_accum = self.weights[:, s_in].sum(axis=1).A1
+        self.a_buff = (
+            np.left_shift(a_accum, self.weight_exp)
+            if self.weight_exp > 0
+            else np.right_shift(a_accum, -self.weight_exp)
+        )
+
+
+@implements(proc=Sparse, protocol=LoihiProtocol)
+@requires(CPU)
+@tag("bit_accurate_loihi", "fixed_pt")
+class PySparseModelBitAcc(AbstractPySparseModelBitAcc):
+    pass
+
+
+@implements(proc=LearningSparse, protocol=LoihiProtocol)
+@requires(CPU)
+@tag("floating_pt")
+class PyLearningSparseModelFloat(
+        LearningConnectionModelFloat, AbstractPySparseModelFloat):
+    """Implementation of Conn Process with Sparse synaptic connections in
+    floating point precision. This short and simple ProcessModel can be used
+    for quick algorithmic prototyping, without engaging with the nuances of a
+    fixed point implementation.
+
+    Warning: LearningSparse on CPU is not offereing any memory usage benefits
+    over using LearningDense.
+    """
+
+    # overwrite dense PyTypes with sparse
+    tag_1: csr_matrix = LavaPyType(csr_matrix, float)
+    tag_2: csr_matrix = LavaPyType(csr_matrix, float)
+
+    def __init__(self, proc_params):
+        super().__init__(proc_params)
+
+    def run_spk(self):
+        # The a_out sent at each timestep is a buffered value from dendritic
+        # accumulation at timestep t-1. This prevents deadlocking in
+        # networks with recurrent connectivity structures.
+        self.a_out.send(self.a_buff)
+        if self.num_message_bits.item() > 0:
+            s_in = self.s_in.recv()
+            self.a_buff = self.weights.dot(s_in)
+        else:
+            s_in = self.s_in.recv().astype(bool)
+            self.a_buff = self.weights[:, s_in].sum(axis=1).A1
+
+        self.recv_traces(s_in)
+
+
+@implements(proc=LearningSparse, protocol=LoihiProtocol)
+@requires(CPU)
+@tag("bit_approximate_loihi", "fixed_pt")
+class PyLearningSparseModelBitApproximate(
+        LearningConnectionModelBitApproximate, AbstractPySparseModelBitAcc):
+    """Implementation of Conn Process with Sparse synaptic connections that is
```

Comment:
See my other comment above and adapt accordingly

Reply:
done